pkg/operator/configobservation/images/observe_images.go

```diff
@@ -52,3 +53,53 @@ func ObserveInternalRegistryHostname(genericListers configobserver.Listers, reco
 
 	return observedConfig, errs
 }
+
+// ObserveExternalRegistryHostnames observers information about registry external URLs,
+// aka Routes. It retrieves this information from cluster's Image config.
+func ObserveExternalRegistryHostnames(
+	genericListers configobserver.Listers,
+	recorder events.Recorder,
+	existingConfig map[string]interface{},
+) (map[string]interface{}, []error) {
+	var errs []error
+	prevObservedConfig := map[string]interface{}{}
+	listers := genericListers.(configobservation.Listers)
+
+	// first observe all the existing config values so that if we get any errors
+	// we can at least return those.
+	cfgpath := []string{"dockerPullSecret", "registryURLs"}
+	currentURLs, _, err := unstructured.NestedStringSlice(existingConfig, cfgpath...)
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
+	if len(currentURLs) > 0 {
+		if err := unstructured.SetNestedStringSlice(
+			prevObservedConfig, currentURLs, cfgpath...,
+		); err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
+	}
+
+	observedConfig := map[string]interface{}{}
+	configImage, err := listers.ImageConfigLister.Get("cluster")
+	if errors.IsNotFound(err) {
+		klog.V(2).Infof("images.config.openshift.io/cluster: not found")
+		return observedConfig, errs
+	} else if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
+
+	sset := sets.NewString()
+	sset.Insert(configImage.Spec.ExternalRegistryHostnames...)
+	sset.Insert(configImage.Status.ExternalRegistryHostnames...)
```

Comment:
Potential things to consider when using both `spec` and `status` to populate the external urls:

1. If the cluster admin adds external hostnames to spec, openshift-controller-manager has a rollout before the external hostname is reflected in the image status. I believe this doesn't happen until the route for the external hostname is created.
2. Likewise if the cluster admin removes external hostnames, openshift-controller-manager will retain "bad" external hostnames as long as the deleted external hostname is in status.

I'm inclined to only use status to set the `registryURLs` field.

Reply:
That is interesting. During my tests I have noticed that the Status external hostname was not being kept in sync with the Spec (was planning to open a BZ for that). Turns out that the software is way clever than I am (I haven't created the route). I will push a change to use only Status. Thanks for that.

Reply:
Done, PTAL.